fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/AggregationAverageMaker.java

```diff
@@ -115,80 +120,69 @@ private TemplateDruidQuery buildOuterQuery(
-        if (sourceMetric instanceof PostAggregation) {
-            newInnerPostAggregations.add((PostAggregation) sourceMetric);
-        }
+        Set<PostAggregation> newInnerPostAggregations = !(sourceMetric instanceof PostAggregation) ?
+                Collections.emptySet() : ImmutableSet.of((PostAggregation) sourceMetric);
 
         // Build the inner query with the new aggregations and with the count
-        TemplateDruidQuery innerQuery = innerDependentQuery;
-        innerQuery = innerQuery.withAggregations(newInnerAggregations);
-        innerQuery = innerQuery.withPostAggregations(newInnerPostAggregations);
-        innerQuery = innerQuery.merge(buildTimeGrainCounterQuery());
-        return innerQuery;
+        return innerDependentQuery.withAggregations(newInnerAggregations)
+                .withPostAggregations(newInnerPostAggregations)
+                .merge(buildTimeGrainCounterQuery());
     }
 
     /**
-     * If the aggregation being averaged is a sketch, it will become a sketch merge, and need to be estimated for
-     * summing to work at the outer level.
+     * If the aggregation being averaged is a sketch, the inner query must convert it to a numerical type so that it
+     * can be summed.
      *
      * @param originalSourceMetric  The metric being target for sums
      *
      * @return Either the original MetricField, or a new SketchEstimate post aggregation
      */
     private MetricField convertToSketchEstimateIfNeeded(MetricField originalSourceMetric) {
-        if (originalSourceMetric instanceof SketchAggregation) {
-            return FieldConverterSupplier.sketchConverter.asSketchEstimate((SketchAggregation) originalSourceMetric);
-        }
-        return originalSourceMetric;
+        return !(originalSourceMetric instanceof SketchAggregation) ?
+                originalSourceMetric :
+                sketchConverter.asSketchEstimate((SketchAggregation) originalSourceMetric);
     }
 
     /**
      * Copy a set of aggregations, replacing any sketch aggregations with sketchMerge aggregations.
+     * This is an artifact of earlier sketch code which didn't have a single sketch type that could be used without
+     * finalization in inner queries.
      *
      * @param originalAggregations  The read-only original aggregations
      *
      * @return The new aggregation set
      */
     private Set<Aggregation> convertSketchesToSketchMerges(Set<Aggregation> originalAggregations) {
-        Set<Aggregation> result = new LinkedHashSet<>();
-        for (Aggregation agg : originalAggregations) {
-            if (agg.isSketch()) {
-                result.add(FieldConverterSupplier.sketchConverter.asInnerSketch((SketchAggregation) agg));
-            } else {
-                result.add(agg);
-            }
-        }
-        return result;
+        return originalAggregations.stream()
+                .map(agg -> ! agg.isSketch() ? agg : sketchConverter.asInnerSketch((SketchAggregation) agg))
```

Comment:
I would flip these conditions, and make the check a positive check rather than a negative one.